src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -13,9 +14,12 @@ namespace Microsoft.AspNetCore.Routing
 {
     internal sealed class EndpointRoutingMiddleware
     {
+        private const string DiagnosticsRouteMatchedKey = "Microsoft.AspNetCore.Routing.RouteMatched";
+
         private readonly MatcherFactory _matcherFactory;
         private readonly ILogger _logger;
         private readonly EndpointDataSource _endpointDataSource;
+        private readonly DiagnosticListener _diagnosticListener;
```

Comment:
I think this should be typed as `DiagnosticSource` instead of `DiagnosticListener`. We're only using the type to raise diagnostic events, not subscribe to them.

When implementing this in gRPC yesterday I looked for the asp.net core source to see what is done here, and there is a mix between using listener and source. It probably doesn't matter, but some consistency might be helpful.

Reply:
Perf!! The listener has the faster IsEnabled check

Reply:
Huh, ok. Why is it on the listener and not the source :\

Reply:
virtual + sealed override.